metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_tag.py

```diff
@@ -0,0 +1,138 @@
+import logging
+from typing import Dict, List, Optional
+
+from datahub.ingestion.source.snowflake.snowflake_config import (
+    SnowflakeV2Config,
+    TagOption,
+)
+from datahub.ingestion.source.snowflake.snowflake_report import SnowflakeV2Report
+from datahub.ingestion.source.snowflake.snowflake_schema import (
+    SnowflakeDataDictionary,
+    SnowflakeTag,
+    _SnowflakeTagCache,
+)
+from datahub.ingestion.source.snowflake.snowflake_utils import SnowflakeCommonMixin
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class SnowflakeTagExtractor(SnowflakeCommonMixin):
+    def __init__(
+        self,
+        config: SnowflakeV2Config,
+        data_dictionary: SnowflakeDataDictionary,
+        report: SnowflakeV2Report,
+    ) -> None:
+        self.config = config
+        self.data_dictionary = data_dictionary
+        self.report = report
+        self.logger = logger
+
+        self.tag_cache: Dict[str, _SnowflakeTagCache] = {}
+
+    def get_tags_on_object(
+        self,
+        domain: str,
+        db_name: str,
+        schema_name: Optional[str] = None,
+        table_name: Optional[str] = None,
+    ) -> List[SnowflakeTag]:
+        if self.config.extract_tags == TagOption.without_lineage:
+            if db_name not in self.tag_cache:
```

Comment:
Can we  refractor contents of if condition into another method.  Say -  `get_tags_on_object_without_lineage`
We are trying to move towards small readable and unit-testable methods.